### PR TITLE
ci: use azure pipelines

### PR DESCRIPTION
**Pull request by james-d-mitchell**

### CODE REVIEW

**Review by wilfwilson — COMMENTED** [review]
My only concern with this PR is that it breaks our Appveyor integration.

(Note that Appveyor is not yet integrated with this new repository, do you mind if I enable it?)

We use Appveyor to test Digraphs in 64-bit Cygwin. The Appveyor setup is defined in `.appveyor.yml`. The integration piggy-backs on the Travis integration: once Appveyor sets everything up, it basically calls:
```
  - bash -lc "cd $APPVEYOR_BUILD_FOLDER && ./scripts/travis-build-dependencies.sh"
  - bash -lc "cd $APPVEYOR_BUILD_FOLDER && ./scripts/travis-build-digraphs.sh"
  - bash -lc "cd $APPVEYOR_BUILD_FOLDER && ./scripts/travis-test.sh"
```
I am hoping that these lines can be replaced with the single line:
```
  - bash -lc "cd $APPVEYOR_BUILD_FOLDER && ./ci/docker-test.sh"
```
to run your new unified script, and thereby maintain Appveyor integration.

If it really is this easy, would you mind doing it? I think it would be worth maintaining a Windows CI test if this is the cost.  (The other 'easy' way to maintain Appveyor support would be to simply not delete the files `scripts/travis-build-dependencies.sh`, `scripts/travis-build-digraphs.sh`, and `scripts/travis-test.sh` to give us some breathing space until either we work out a proper Windows-based solution, or we decide to ignore Windows.)

**Comment by wilfwilson** [comment]
(And thanks for doing this, by the way)

**Comment by james-d-mitchell** [comment]
Thanks for the reminder @wilfwilson I'm happy to readd appveyor integration. It won't be as simple as just adding 

~~~~
  - bash -lc "cd $APPVEYOR_BUILD_FOLDER && ./ci/docker-test.sh"
~~~~

because the `docker-test.sh` doesn't contain anything for getting or building GAP itself (this is installed in the docker image already). There's no point in running tests on appveyor inside a docker container (because this is just the same as running it on unix). I'll have to think about how to resolve this properly.

**Comment by wilfwilson** [comment]
Ah yes, of course, that makes sense.

For now, I recommend the safe option would be to remove the deletion of the three `scripts/travis-*.sh` files from this PR. And we can deal with removing or replacing the Windows tests in a different PR.

**Comment by james-d-mitchell** [comment]
Yes, I'll do what you suggest

**Comment by james-d-mitchell** [comment]
I re-added the deleted files and I think I enable appveyor again but nothing has been triggered, and ideas?

File: scripts/travis-build-dependencies.sh
Comment (wilfwilson) [inline scripts/travis-build-dependencies.sh:113]: Not a huge problem, but could you change this back to `gap-packages` please?

To answer your question: I hadn't enabled Appveyor on this new repository yet, but I have done so now, so it should run in future (as is indeed happening over here https://github.com/digraphs/Digraphs/pull/347).

If you're ready for this to be merged (I'm happy to approve), then would you mind squashing the commits? I could do it with a squash and merge, but I'd feel happier to have all the CI (including Appveyor) run through fully once in the squashed form.

(Since approved reviews get dismissed after each push, I'll hold off approving till I actually merge).

Comment (wilfwilson) [reply scripts/travis-build-dependencies.sh:113]: And can you change the occurrence of `gap-packages` in the file `scripts/travis-build-digraphs.sh` to be `digraphs`? I'm not going to insist that Appveyor works before merging this PR, but I'm pretty sure that if you make that change, then Appveyor test will pass (see https://ci.appveyor.com/project/wilfwilson/digraphs/builds/36359419 for an example)

Comment (james-d-mitchell) [reply scripts/travis-build-dependencies.sh:113]: Will do! Was teaching all day, will fix first thing tomorrow or later tonight

Comment (james-d-mitchell) [reply scripts/travis-build-dependencies.sh]: I think I made the requested changes. Hopefully everything will pass !

**Comment by james-d-mitchell** [comment]
@wilfwilson appveyor has failed, did I miss something?

**Review by wilfwilson — APPROVED** [review]
No, you did everything I asked, thank you.

I've worked out why AppVeyor still failed (even though it passed on my fork) - I'll explain below. I've made a PR (https://github.com/digraphs/Digraphs/pull/351) to restore AppVeyor.

What happened is that when I set up the AppVeyor integration for this new repository, for some reason (I don't understand how AppVeyor works) AppVeyor automatically created a new 'Project' for running builds for this repository, located at https://ci.appveyor.com/project/james-d-mitchell/digraphs-0gul0/history (I'm not sure why it lives under your username... that's a problem for another day) even though one already existed for builds from `gap-packages/Digraphs` at https://ci.appveyor.com/project/james-d-mitchell/digraphs/history.

Because there was already a so-called 'project' for digraphs, this new automatically-created project got the auto-generated name `digraphs-0gul0`. AppVeyor initially clones Digraphs into a folder with the same name as the project. The `scripts/travis-build-dependencies.sh` script relied on this folder being called `digraphs` (which it still is on my fork - https://ci.appveyor.com/project/wilfwilson/digraphs/builds - and which is why AppVeyor builds were still passing there). But in this new 'Project', the folder is called `digraphs-0gul0`, and so the script fails.

**Pull request merged**
